scrapy/statscollectors.py

```diff
@@ -50,6 +50,15 @@ def close_spider(self, spider, reason):
     def _persist_stats(self, stats, spider):
         pass
 
+    def __getitem__(self, key):
+        return self._stats.get(key)
```

Comment:
I think it makes more sense to raise an exception if a key is not found, like dicts do.

Reply:
Hi, @kmike 

Thanks for your suggestion, I will add that feature to `__getitem__`.

Reply:
> Now, what should be expected when a key is missing? I think to return 0 and create the missing key at the same time could be more consistent with previous behaviors (a debug level logging message could be added here).

I agree that returning 0 and creating a key is convenient in many cases, but if we're implementing a dict-like interface, we better follow dict semantics. Users still have all existing methods available, and they can use them if these methods are a better fit. Implementing `setdefault` method could be better than changing `__getitem__` semantics, as one can write `value = stats.setdefault('my/stats', 0)` to return 0 and create a new key if it is not present, just like with regular Python dicts. It also allows to use non-numbers as default values (not all stats values are numbers).

> Also in this class StatsCollector, when I am going through the source code, I find the behavior of this class is like a dict - why not just inherit the class UserDict in the module collections, and add the necessary methods?

UserDict is removed in Python 3, but inheriting from MutableMapping is a good idea.